src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+		 * @param WP_REST_Request   $request        The request sent to the API.
+		 */
+		do_action( 'bp_members_notices_rest_update_item', $updated_notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request has access to update a notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 */
+	public function update_item_permissions_check( $request ) {
+		$retval = $this->manage_item_permissions_check( $request );
+
+		/**
+		 * Filter the notices `update_item` permissions check.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param bool|WP_Error   $retval  Returned value.
+		 * @param WP_REST_Request $request The request sent to the API.
+		 */
+		return apply_filters( 'bp_members_notices_rest_update_item_permissions_check', $retval, $request );
+	}
+
+	/**
+	 * Dismisses the currently active notice for the current user.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function dismiss_notice( $request ) {
+		$notice_id = $request->get_param( 'id' );
+
+		// Mark the requested notice as closed.
+		if ( $notice_id ) {
+			$notice = bp_members_get_notice( $notice_id );
+
+			// Mark the first priority notice as closed.
+		} else {
+			$notice = bp_get_active_notice_for_user();
+		}
+
+		if ( is_null( $notice ) ) {
+			return new WP_Error(
+				'bp_rest_invalid_id',
+				__( 'Sorry, this notice does not exist.', 'buddypress' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		// Get Previous active notice.
+		$previous = $this->prepare_item_for_response( $notice, $request );
+
+		// Dismiss the active notice for the current user.
+		$dismissed = bp_members_dismiss_notice( bp_loggedin_user_id(), $notice->id );
+
+		// Build the response.
+		$response = new WP_REST_Response();
+		$response->set_data(
+			array(
+				'dismissed' => $dismissed,
+				'previous'  => $previous->get_data(),
+			)
+		);
+
+		/**
+		 * Fires after a notice is dismissed via the REST API.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice   Notice object.
+		 * @param WP_REST_Response  $response The response data.
+		 * @param WP_REST_Request   $request  The request sent to the API.
+		 */
+		do_action( 'bp_members_notices_rest_dismiss_notice', $notice, $response, $request );
+
+		return $response;
+	}
+
+	/**
+	 * Check if a given request has access to dismiss the current notice.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool
+	 */
+	public function dismiss_notice_permissions_check( $request ) {
+		$retval = new WP_Error(
+			'bp_rest_authorization_required',
+			__( 'Sorry, you are not allowed to dismiss notices.', 'buddypress' ),
+			array(
+				'status' => rest_authorization_required_code(),
+			)
+		);
+
+		if ( is_user_logged_in() ) {
```

Comment:
I _think_ it is technically possible for _any_ logged in user to dismiss someone else's notices. As long as they have access to the notice ID. Is this expected?

Reply:
Notices were targeting every member of the community in the previous implementation, I agree I now need to be me restrictive here to check admin notices can only be viewed by admins and contributor notices can only be viewed by contributors.

Reply:
I've updated this part this way:
```php
if ( is_user_logged_in() ) {
		// Users can only get notices targeting them.
		if ( 'view' === $context && bp_loggedin_user_id() === (int) $request->get_param( 'user_id' ) ) {
```